gen2-pointcloud/save_point_cloud/main.py

```diff
@@ -0,0 +1,232 @@
+    """
+    Update the rgb and depth weights used to blend depth/rgb image
+
+    @param[in] percent_rgb The rgb weight expressed as a percentage (0..100)
+    """
+    global depthWeight
+    global rgbWeight
+    rgbWeight = float(percent_rgb) / 100.0
+    depthWeight = 1.0 - rgbWeight
+
+
+# Optional. If set (True), the ColorCamera is downscaled from 1080p to 720p.
+# Otherwise (False), the aligned depth is automatically upscaled to 1080p
+downscaleColor = True
+fps = 30
+# The disparity is computed at this resolution, then upscaled to RGB resolution
+monoResolution = dai.MonoCameraProperties.SensorResolution.THE_720_P
+# widhth and height that match `monoResolution` above
+width = 1280
+height = 720
+# max depth (only used for visualisation) in millimeters
+visualisation_max_depth = 4000
+
+serials = []
+for device in dai.Device.getAllAvailableDevices():
+    serials.append(device.getMxId())
+    print(f"{len(serials)}: {device.getMxId()} {device.state}")
+
+# Create pipeline
+pipeline = dai.Pipeline()
+device = dai.Device()
+queueNames = []
+
+# Define sources and outputs
+camRgb = pipeline.create(dai.node.ColorCamera)
+left = pipeline.create(dai.node.MonoCamera)
+right = pipeline.create(dai.node.MonoCamera)
+stereo = pipeline.create(dai.node.StereoDepth)
+
+rgbOut = pipeline.create(dai.node.XLinkOut)
+depthOut = pipeline.create(dai.node.XLinkOut)
+
+
+rgbOut.setStreamName("rgb")
+queueNames.append("rgb")
+depthOut.setStreamName("depth")
+queueNames.append("depth")
+
+# Properties
+camRgb.setBoardSocket(dai.CameraBoardSocket.RGB)
+camRgb.setResolution(dai.ColorCameraProperties.SensorResolution.THE_1080_P)
+camRgb.setFps(fps)
+if downscaleColor:
+    camRgb.setIspScale(2, 3)
+# For now, RGB needs fixed focus to properly align with depth.
+# This value was used during calibration
+try:
+    calibData = device.readCalibration2()
+    lensPosition = calibData.getLensPosition(dai.CameraBoardSocket.RGB)
+    if lensPosition:
+        camRgb.initialControl.setManualFocus(lensPosition)
+except:
+    raise
+    
+left.setResolution(monoResolution)
+left.setBoardSocket(dai.CameraBoardSocket.LEFT)
+left.setFps(fps)
+right.setResolution(monoResolution)
+right.setBoardSocket(dai.CameraBoardSocket.RIGHT)
+right.setFps(fps)
+
+stereo.setDefaultProfilePreset(dai.node.StereoDepth.PresetMode.HIGH_DENSITY)
+# LR-check is required for depth alignment
+stereo.setLeftRightCheck(True)
+stereo.setDepthAlign(dai.CameraBoardSocket.RGB)
+
+# Linking
+camRgb.isp.link(rgbOut.input)
+left.out.link(stereo.left)
+right.out.link(stereo.right)
+stereo.depth.link(depthOut.input)
+# rectified stereo
+xoutRectifLeft = pipeline.create(dai.node.XLinkOut)
+xoutRectifRight = pipeline.create(dai.node.XLinkOut)
+xoutRectifLeft.setStreamName("rectifiedLeft")
+xoutRectifRight.setStreamName("rectifiedRight")
+stereo.rectifiedLeft.link(xoutRectifLeft.input)
+stereo.rectifiedRight.link(xoutRectifRight.input)
+
+# Connect to device and start pipeline
+with device:
```

Comment:
since you are using context manager, move `with dai.Device(pipeline) as device` here (like in all other demos)